src/main/java/seedu/duke/util/Parser.java

```diff
@@ -401,12 +403,42 @@ private Command prepareAddEvent(String userMessage) {
 
     private Command prepareListEvent(String userMessage) {
         return new ListEventCommand();
-    }
+    }*/
 
-    private Command prepareViewNote(String userMessage) {
-       return new ViewNoteCommand();
-    }
+    private Command prepareViewNote(String userMessage) throws SystemException {
+        String title;
+        int index;
+
+        try {
+            ArrayList<String[]> splitInfo = splitInfoDetails(userMessage);
+
+            for (String[] infoDetails : splitInfo) {
+                String prefix = infoDetails[0];
+                if (infoDetails[1].isBlank()) {
+                    throw new SystemException(SystemException.ExceptionType.EXCEPTION_MISSING_DESCRIPTION);
+                }
+                switch (prefix) {
+                case PREFIX_TITLE:
+                    title = infoDetails[1].trim();
+                    return new ViewNoteCommand(title);
+                case PREFIX_INDEX:
+                    index = Integer.parseInt(infoDetails[1].trim());
+                    return new ViewNoteCommand(index);
+                default:
+                    throw new SystemException(SystemException.ExceptionType.EXCEPTION_MISSING_TAG);
+                }
+            }
+        } catch (NullPointerException exception) {
+            throw new SystemException(ExceptionType.EXCEPTION_MISSING_TAG_PREFIX);
+        } catch (ArrayIndexOutOfBoundsException exception) {
+            throw new SystemException(ExceptionType.EXCEPTION_INVALID_INDEX_VALUE);
+        } catch (NumberFormatException exception) {
+            throw new SystemException(ExceptionType.EXCEPTION_INVALID_INDEX_FORMAT);
+        }
 
+        return new IncorrectCommand("Incorrect format for viewing notes");
```

Comment:
Would it ever reach this point? If so, can change it to throw SystemException instead of return new IncorrectCommand for consistency with other prepare functions.